src/ResourceManager/IotHub/Commands.IotHub/IotHub/GetAzureRmIotHubCertificate.cs

```diff
@@ -0,0 +1,110 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+    using ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet(VerbsCommon.Get, "AzureRmIotHubCertificate", DefaultParameterSetName = ResourceParameterSet)]
```

Comment:
For Get-Certificate, it also only makes sense to pipe in the IotHub. If you had the certificate object, you will not need to run Get-Certificate.

Reply:
True, but if you wont pass certificatename then it will work as get-certificates otherwise get-certificate

Reply:
Yes, but you won't be able to pipe Get-AzureRmIotHub | Get-AzureRmIotHubCertificate because the InputObject only accepts a PSCertificateDescription, which won't make sense in most piping situations.

Reply:
updated this scenario,

Reply:
This isn't what I intended - remove the valuefrompipelinebypropertyname, and change inputobject to iotHub and resourceId to IotHub resourceId